src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -124,19 +151,381 @@ public static IResourceBuilder<AzureServiceBusResource> AddQueue(this IResourceB
+
+        var customMountAnnotation = new ContainerMountAnnotation(
+                configHostFile,
+                AzureServiceBusEmulatorResource.EmulatorConfigJsonPath,
+                ContainerMountType.BindMount,
+                isReadOnly: false);
+
+        builder
+            .WithEndpoint(name: "emulator", targetPort: 5672)
+            .WithAnnotation(new ContainerImageAnnotation
+            {
+                Registry = ServiceBusEmulatorContainerImageTags.Registry,
+                Image = ServiceBusEmulatorContainerImageTags.Image,
+                Tag = ServiceBusEmulatorContainerImageTags.Tag
+            })
+            .WithAnnotation(customMountAnnotation);
+
+        var sqlEdgeResource = builder.ApplicationBuilder
+                .AddContainer($"{builder.Resource.Name}-sqledge",
+                    image: ServiceBusEmulatorContainerImageTags.AzureSqlEdgeImage,
+                    tag: ServiceBusEmulatorContainerImageTags.AzureSqlEdgeTag)
+                .WithImageRegistry(ServiceBusEmulatorContainerImageTags.AzureSqlEdgeRegistry)
+                .WithEndpoint(targetPort: 1433, name: "tcp")
+                .WithEnvironment("ACCEPT_EULA", "Y")
+                .WithEnvironment("MSSQL_SA_PASSWORD", password);
+
+        builder.WithAnnotation(new EnvironmentCallbackAnnotation((EnvironmentCallbackContext context) =>
+        {
+            var sqlEndpoint = sqlEdgeResource.Resource.GetEndpoint("tcp");
+
+            context.EnvironmentVariables.Add("ACCEPT_EULA", "Y");
+            context.EnvironmentVariables.Add("SQL_SERVER", $"{sqlEndpoint.Resource.Name}:{sqlEndpoint.TargetPort}");
+            context.EnvironmentVariables.Add("MSSQL_SA_PASSWORD", password);
+        }));
+
+        string? connectionString = null;
+        string[]? queueNames = null;
+        string[]? topicNames = null;
+
+        builder.ApplicationBuilder.Eventing.Subscribe<ConnectionStringAvailableEvent>(builder.Resource, async (@event, ct) =>
+        {
+            var serviceBusEmulatorResources = builder.ApplicationBuilder.Resources.OfType<AzureServiceBusResource>().Where(x => x is { } serviceBusResource && serviceBusResource.IsEmulator);
+
+            if (!serviceBusEmulatorResources.Any())
+            {
+                // No-op if there is no Azure Service Bus emulator resource.
+                return;
+            }
+
+            connectionString = await builder.Resource.ConnectionStringExpression.GetValueAsync(ct).ConfigureAwait(false);
+
+            if (connectionString == null)
+            {
+                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{builder.Resource.Name}' resource but the connection string was null.");
+            }
+
+            queueNames = serviceBusEmulatorResources.SelectMany(x => x.Queues).Select(x => x.Name).ToArray();
+            topicNames = serviceBusEmulatorResources.SelectMany(x => x.Topics).Select(x => x.Name).ToArray();
```

Comment:
Do we really need to check every queue and topic each time the health check is run? Don't we really only need to check 1?

Reply:
Sure, we could expect everything to be ready once one is available, but I didn't want to make that assumption. The implementation in the nuget package is per queue/topic, so it doesn't give that info. Have asked the team for now.